cfy_cluster_manager/main.py

```diff
@@ -195,51 +205,162 @@ def _prepare_config_files(instances_dict,
 
 
 def _install_cloudify_remotely(instance, rpm_download_link):
+    logger.info('Downloading Cloudify RPM on %s from %s',
+                instance.name, rpm_download_link)
+    instance.run_command('curl -o {0} {1}'.format(
+        RPM_PATH, rpm_download_link), hide_stdout=True)
+    logger.info('Installing Cloudify RPM on %s', instance.name)
+    instance.run_command(
+        'yum install -y {}'.format(RPM_PATH), use_sudo=True, hide_stdout=True)
+
+
+def _get_service_status_code(instance):
+    """Checking the status code of the cfy_manager_install_<type> service."""
+    result = instance.run_command(
+        'systemctl status {}'.format(instance.unit_name),
+        use_sudo=True, hide_stdout=True, ignore_failure=True)
+    return result.return_code
+
+
+def _wait_for_cloudify_current_installation(instance):
+    logger.info(
+        'Waiting for current installation of %s to finish', instance.name)
+    status_code = 0
+    timeout = 0
+    while status_code == 0 and timeout <= 600:
+        if timeout % 20 == 0:
+            logger.info('Waiting for current installation of %s to finish',
+                        instance.name)
+        time.sleep(1)
+        timeout += 1
+        status_code = _get_service_status_code(instance)
+
+    if timeout > 600:
+        raise ClusterInstallError(
+            'Got a time out while waiting for the current installation of '
+            '%s to finish', instance.name)
+
+
+def _rpm_was_installed(instance, rpm_download_link):
     rpm_name = splitext(basename(rpm_download_link))[0]
-    try:
-        instance.run_command('rpm -qa | grep {0}'.format(rpm_name),
-                             hide_stdout=True)
-        logger.info('Cloudify RPM is already installed on %s',
-                    instance.name)
-    except ClusterInstallError:
-        # If cloudify is not installed, an error is raised
-        logger.info('Downloading Cloudify RPM on %s from %s',
-                    instance.name, rpm_download_link)
-        instance.run_command('curl -o {0} {1}'.format(
-            RPM_PATH, rpm_download_link))
-        logger.info('Installing Cloudify RPM on %s', instance.name)
+    logger.debug(
+        'Checking if Cloudify RPM was installed on %s', instance.private_ip)
+    result = instance.run_command('rpm -qa | grep cloudify-manager-install',
```

Comment:
I think you can just do this with `rpm -qi cloudify-manager-install` and then check the return code?

Reply:
Right, changed.